### PR TITLE
check-pause-resume: add support for SOF_ALSA/APLAY/ARECORD_OPTS

### PR DESCRIPTION
**Pull request by marc-hb**
2 commits: 1. shellcheck clean-up, 2. SOF_ALSA_OPTS

### CODE REVIEW

**Comment by marc-hb** [comment]
On my APL_UP2_NOCODEC I get this error message from time to time:
```
(1/10) Wait for 155 ms before pause                | 00%
=== PAUSE ===           aplay: do_pause:1545: pause push error: File descriptor in bad state
(1/10) Wait for 197 ms before pause                | 00%                  
(1/10) Wait for 116 ms before pause                                      
(1/10) Wait for 118 ms before resume
```

Yet `SOF_ALSA_OPTS='--fatal-errors'`  does NOT stop `aplay` after it prints this error and it does not make the test fail #greenfailures

cc: @plbossart

**Comment by marc-hb** [comment]
Pure rebase (after # 529), no code change.

File: test-case/check-pause-resume.sh [hunk 1]
```diff
@@ -75,24 +78,25 @@ esac
...
+    rate=$(func_pipeline_parse_value "$idx" rate)
+    fmt=$(func_pipeline_parse_value "$idx" fmt)
+    dev=$(func_pipeline_parse_value "$idx" dev)
+    snd=$(func_pipeline_parse_value "$idx" snd)
 
     # expect is tcl language script
     #   expr rand(): produces random numbers between 0 and 1
     #   after ms: Ms must be an integer giving a time in milliseconds.
     #       The command sleeps for ms milliseconds and then returns.
+    dlogi "Entering expect script with:
```
Comment (xiulipan) [inline test-case/check-pause-resume.sh:95]: @marc-hb I see we have duplicated logs, example:

https://sof-ci.01.org/softestpr/PR527/build423/devicetest/?model=ICL_RVP_HDA&testcase=check-pause-resume-playback-10
```
2020-11-23 22:32:27 UTC [REMOTE_INFO] Entering expect script with:
      aplay   -D hw:0,5 -r 48000 -c 2 -f S16_LE -vv -i /dev/zero -q
spawn aplay -D hw:0,5 -r 48000 -c 2 -f S16_LE -vv -i /dev/zero -q
```

**Review by xiulipan — APPROVED** [review]
Looks good, but we seems to have duplicated logs output with expect.

File: test-case/check-pause-resume.sh (hunk 1 above)
Comment (marc-hb) [reply test-case/check-pause-resume.sh:95]: The log was already duplicated, I just added a newline that made it more obvious. Before:

```shell
git checkout origin/master
./test-case/check-pause-resume.sh

2020-12-02 04:25:33 UTC [INFO] Entering expect script with: aplay -D hw:0,0 -r 48000 -c 2 -f S16_LE -vv -i /dev/zero -q
spawn aplay -D hw:0,0 -r 48000 -c 2 -f S16_LE -vv -i /dev/zero -q
```
I can easily escape the newline so the output is exactly like before.

I'm a bit reluctant to trust and rely entirely on `expect` for the logging... maybe that's also why someone else doubled this?

Comment (xiulipan) [reply test-case/check-pause-resume.sh:95]: Good point here, let's compare the cmd here with both logs.

**Comment by marc-hb** [comment]
For the record this should help with #329 and #489

**Pull request merged**